internal/akams/akams.go

```diff
@@ -140,49 +172,101 @@ func (c *Client) CreateOrUpdateBulk(ctx context.Context, links []CreateLinkReque
 	return nil
 }
 
-// UpdateBulk updates multiple links in bulk.
-func (c *Client) UpdateBulk(ctx context.Context, links []UpdateLinkRequest) error {
-	req, err := c.newRequest(ctx, http.MethodPut, "bulk", links)
+func (c *Client) exists(ctx context.Context, shortURL string) (bool, error) {
+	req, err := c.newRequest(ctx, http.MethodGet, shortURL, nil)
 	if err != nil {
-		return fmt.Errorf("failed to create request: %v", err)
+		return false, err
 	}
-	resp, err := c.httpClient.Do(req)
+	resp, err := c.do(req)
 	if err != nil {
-		return fmt.Errorf("request failed: %v", err)
+		return false, err
 	}
 	defer resp.Body.Close()
 	switch resp.StatusCode {
-	case http.StatusAccepted, http.StatusNoContent, http.StatusNotFound:
-		// success
+	case http.StatusOK:
+		return true, nil
+	case http.StatusNotFound:
+		return false, nil
 	default:
-		return c.reqError(resp)
+		return false, c.reqError(resp)
 	}
-	return nil
 }
 
 func (c *Client) reqError(resp *http.Response) error {
+	// Try to preserve the response body. It may have important context to fix the issue.
 	body, _ := io.ReadAll(resp.Body)
 	return &ResponseError{StatusCode: resp.StatusCode, Body: string(body)}
 }
 
-func (c *Client) newRequest(ctx context.Context, method string, urlStr string, body any) (*http.Request, error) {
+func (c *Client) newRequest(ctx context.Context, method string, urlStr string, body io.Reader) (*http.Request, error) {
 	u := c.baseURL.JoinPath(urlStr)
-	var buf io.ReadWriter
-	if body != nil {
-		buf = &bytes.Buffer{}
-		enc := json.NewEncoder(buf)
-		enc.SetEscapeHTML(false)
-		if err := enc.Encode(body); err != nil {
-			return nil, err
-		}
-	}
-	req, err := http.NewRequestWithContext(ctx, method, u.String(), buf)
+	req, err := http.NewRequestWithContext(ctx, method, u.String(), body)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to create request: %v", err)
 	}
 	req.Header.Set("Accept", "application/json")
 	if body != nil {
 		req.Header.Set("Content-Type", "application/json")
 	}
 	return req, nil
 }
+
+func (c *Client) do(req *http.Request) (*http.Response, error) {
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("request failed: %v", err)
+	}
+	return resp, nil
+}
+
+// chunkSlice chunks s into bulkSize chunks preserving the order of the items,
+// encodes each chunk into JSON, and calls fn with the encoded chunk.
+// The maximum encoded size of each chunk is limited to maxSizeBytes.
```

Comment:
I started putting together a suggestion to clarify that the first line doesn't just mean "into N chunks" where N = bulkSize, but ended up moving some other things around to (IMO) make it more digestible:

```suggestion
// chunkSlice chunks s, encodes each chunk into JSON,
// and calls fn with the encoded chunk.
// The maximum encoded size of each chunk is limited to maxSizeBytes,
// and each chunk has at most bulkSize items.
// The order of the items is preserved.
```